tests/async_/test_async_transport.py

```diff
@@ -45,6 +45,14 @@
 from tests.conftest import AsyncDummyNode
 
 
+def exception_to_dict(exc: TransportError) -> dict:
+    return {
+        "type": exc.__class__.__name__,
+        "message": exc.message,
+        "errors": [exception_to_dict(e) for e in exc.errors],
+    }
```

Comment:
This is all the information exposed by `__repr__`, but in a more structured way which is nicer to test for.